### DISCUSSION TITLE
where to put devtools middleware?

### DISCUSSION THREAD

**Discussion by ghost** [pmndrs/zustand#1949] [Q&A]
### This part of documentation confused me. Sample code uses devtool middleware as last one as it is recommended. Then it says we should write immer(devtools(... ,  not    devtools(immer(...
Could you please clarify it?
![image](https://github.com/pmndrs/zustand/assets/138553763/acee64d9-81ea-452b-8d86-8b9e378c3dd5)

**Comment by dai-shi** [comment]
The working combinations are in the `middlewareTypes.test.tsx` file
 https://github.com/pmndrs/zustand/blob/f11cc7f37ad0a53f49370c0e4e659690a1c1a721/tests/middlewareTypes.test.tsx#L380

I'm not sure, but it looks like the docs are misleading.

**Comment by JMY1993** [comment]
I believe it was a typo and they have changed their doc
![image](https://github.com/user-attachments/assets/c8d6aee4-5cb6-4878-a20d-2edb01bb6ef8)
